modules/tinymce/src/plugins/codesample/test/ts/browser/CodeSampleSanityTest.ts

```diff
@@ -41,4 +41,12 @@ describe('browser.tinymce.plugins.codesample.CodeSampleSanityTest', () => {
     await TestUtils.pCancelDialog(editor);
     TinyAssertions.assertContent(editor, '');
   });
+
+  it('TINY-8861: set content places the cursor in a valid position', () => {
+    const editor = hook.editor();
+
+    editor.setContent('<pre class="language-markup"><code>test content</code></pre>');
+    TinyAssertions.assertCursor(editor, [ 0 ], 0);
```

Comment:
Why do we place the cursor at the beginning but not the end of the code sample?

Reply:
Because this is how the `setContent` API logic works today 🤷 https://fiddle.tiny.cloud/ziiaab